runtime/cmake/platform.cmake

```diff
@@ -35,8 +35,8 @@ if(OMR_TOOLCONFIG STREQUAL "gnu")
     set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -pthread -O3 -fno-strict-aliasing")
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -pthread -O3 -fno-strict-aliasing -fno-exceptions -fno-rtti -fno-threadsafe-statics")
 elseif(OMR_TOOLCONFIG STREQUAL "xlc")
-    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -O3 -qalias=noansi")
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -O3 -qalias=noansi -qnortti -qnoeh -qsuppress=1540-1087:1540-1088:1540-1090")
+    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -O3 -g -qalias=noansi")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -O3 -g -qalias=noansi -qnortti -qnoeh -qsuppress=1540-1087:1540-1088:1540-1090")
```

Comment:
The uma makefiles seem to handle this differently and only enable based on `j9vm_uma_gnuDebugSymbols`

OSX:
https://github.com/eclipse/openj9/blob/2321af3101b8d926f801fe134fc63e0d5aadc1c6/runtime/makelib/targets.mk.osx.inc.ftl#L92-L95
https://github.com/eclipse/openj9/blob/2321af3101b8d926f801fe134fc63e0d5aadc1c6/runtime/makelib/targets.mk.osx.inc.ftl#L131-L133

AIX (note, it disables `-g` for the bytecode interpreter:
https://github.com/eclipse/openj9/blob/b8e84f7c926b77617c11ecc1c1c0dffd6b759d0b/runtime/makelib/targets.mk.aix.inc.ftl#L82-L88

linux
https://github.com/eclipse/openj9/blob/13b6e3f0bc47df4fca2eef35f8bac9e6ab560958/runtime/makelib/targets.mk.linux.inc.ftl#L236-L247

Reply:
This is true, however I believe that j9vm_uma_gnuDebugSymbols is always enabled

Reply:
It makes reviewing the code much simpler if it's directly ported rather than changed during the port as we can see the conditions are the same.

The overall goal of removing always-on flags is good but would be clearer if handled as a separate item